app/models/account.rb

```diff
@@ -495,8 +495,8 @@ def to_xml(options = {})
         fields_to_xml(xml)
         extra_fields_to_xml(xml)
 
-        xml.monthly_billing_enabled settings.monthly_billing_enabled
-        xml.monthly_charging_enabled settings.monthly_charging_enabled
+        xml.monthly_billing_enabled settings&.monthly_billing_enabled
```

Comment:
I think these nil-checks that we are having to inject here all over are too subtle. They are meant for the edge case of the record getting accessed while in the process of being deleted. One who missed this PR and may later read the code and think it's expected to be nil.

I think it deserves at least a comment, but better somehow being explicit about the state of the record and why it doesn't exist in this case. Something to think about.